core/src/autogluon/core/scheduler/scheduler_constructor.py

```diff
@@ -0,0 +1,131 @@
+import logging
+
+from ..task.base import compile_scheduler_options_v2
+from ..task.base.base_task import schedulers
+from ..utils.utils import setup_compute
+
+logger = logging.getLogger()
+
+
+_scheduler_presets = {
+    'auto': {'searcher': 'random'},
+    # 'grid': {'searcher': 'grid'},  # grid commented out as it isn't compatible with most default model search spaces
+    'random': {'searcher': 'random'},
+    'bayesopt': {'searcher': 'bayesopt'},
+    # 'skopt': {'searcher': 'skopt'},  # TODO: Remove skopt? Is it worthwhile to keep as an option?
+    # Don't include hyperband and bayesopt hyperband at present
+}
+
+
+def init_scheduler(
+        hyperparameter_tune_kwargs,
+        time_out: float = None,
+        num_trials: int = None,
+        nthreads_per_trial='all',
+        ngpus_per_trial='all',
```

Comment:
Would we ever have other types of resources constraints? I was exploring HPO tuners and there's usually generic 'resources' bucket where cpu/gpu just few of these with options to plug custom resources.

Reply:
Possibly, although generalizing in this fashion makes the API more confusing, so its a tradeoff. I'd recommend we keep it as is for now and add additional options when they become relevant.